Grbl_Esp32/Custom/CoreXY.cpp

```diff
@@ -147,9 +147,9 @@ bool user_defined_homing(uint8_t cycle_mask) {
                         }
                         st_prep_buffer();  // Check and prep segment buffer. NOTE: Should take no longer than 200us.
                         // Exit routines: No time to run protocol_execute_realtime() in this loop.
-                        if (sys_rt_exec_state.bit.safetyDoor || sys_rt_exec_state.bit.reset || cycle_stop) {
+                        if (sys_safetyDoor || sys_reset || sys_cycleStop) {
                             ExecState rt_exec_state;
-                            rt_exec_state.value = sys_rt_exec_state.value;
+                            rt_exec_state.value = sys_get_rt_exec_state().value;
```

Comment:
I wonder if we really need to collect all the bits into one value.  Can we eliminate sys_get_rt_exec_state() entirely and just use the individual variables directly?  Have you identified a specific race condition that the collection solves?